File: sound-parent/sound-web/src/main/resources/static/js/controller/groupController.js
 //控制层 
app.controller('groupController' ,function($scope,$controller   ,groupService){	
	
	$controller('baseController',{$scope:$scope});//继承


    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		$scope.groupList={data:[]};//父组id列表
		groupService.selectForSelect2().success(
			function(response){
				$scope.groupList={data:response};
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		groupService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	}
	
	//查询实体 
	$scope.findOne=function(id){				
		groupService.findOne(id).success(
			function(response){
				$scope.entity= response;
				$scope.entity.groupParentid=JSON.parse($scope.entity.groupParentid);

			}
		);				
	}
	
	//保存 
	$scope.save=function(){				
		var serviceObject;//服务层对象
		$scope.entity.groupParentid=JSON.stringify($scope.entity.groupParentid);
		if($scope.entity.groupId!=null){//如果有ID
			serviceObject=groupService.update( $scope.entity ); //修改  
		}else{

			serviceObject=groupService.add( $scope.entity  );//增加 
		}				
		serviceObject.success(
			function(response){
				if(response.success){
					//重新查询 
		        	$scope.reloadList();//重新加载
				}else{
					alert(response.message);
				}
			}		
		);				
	}
	
	 
	//批量删除 
	$scope.dele=function(){			
		//获取选中的复选框			
		groupService.dele( $scope.selectIds ).success(
			function(response){
				if(response.success){
					$scope.reloadList();//刷新列表
					$scope.selectIds=[];
				}						
			}		
		);				
	}
	
	$scope.searchEntity={};//定义搜索对象 
	
	//搜索
	 $scope.search=function(page,rows){
	 	groupService.search(page,rows,$scope.searchEntity).success(
	 		function(response){
	 			$scope.list=response.rows;
	 			$scope.paginationConf.totalItems=response.total;//更新总记录数
			}
	 	);
	 }

});	

File: sound-parent/sound-web/src/main/java/com/wisdom/soundweb/controller/GroupController.java
package com.wisdom.soundweb.controller;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import com.wisdom.sound.entity.PageResult;
import com.wisdom.sound.entity.Result;
import com.wisdom.sound.pojo.Group;
import com.wisdom.util.numutil.UUIDutil;
import com.wisdom.util.numutil.dateUtil;
import org.apache.dubbo.config.annotation.DubboReference;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

import com.wisdom.sound.service.GroupService;


/**
 * controller
 * @author Administrator
 *
 */
@Controller
@RequestMapping("group")
public class GroupController {

	@DubboReference
	private GroupService GroupServiceImpl;
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findAll")
	@ResponseBody
	public List<Group> findAll(){
		return GroupServiceImpl.findAll();
	}
	@RequestMapping("/groupList")
	@ResponseBody
	public  List<HashMap<String,String>> groupList(){

		List<HashMap<String,String>> groups = GroupServiceImpl.selectForSelect2();
		return groups;
	}

//	@RequestMapping("/test")
//	public Group test(){
//		return GroupServiceImpl.test();
//	}

	@RequestMapping("/togroup")
	public String togroup(){
		return "/admin/group";
	}

	@RequestMapping("/tohome")
	public String tohome(){
		return "/admin/home";
	}

	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findPage")
	@ResponseBody
	public PageResult findPage(int page, int rows){

		return GroupServiceImpl.findPage(page, rows);
	}
	
	/**
	 * 增加
	 * @param group
	 * @return
	 */
	@RequestMapping("/add")
	@ResponseBody
	public Result add(@RequestBody Group group){
		try {
			group.setGroupId(UUIDutil.getUUID());
			group.setGroupCreattime(dateUtil.getDate());

			GroupServiceImpl.add(group);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}
	
	/**
	 * 修改
	 * @param group
	 * @return
	 */
	@RequestMapping("/update")
	@ResponseBody
	public Result update(@RequestBody Group group){
		try {
			GroupServiceImpl.update(group);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	/**
	 * 获取实体
	 * @param
	 * @return
	 */
	@RequestMapping("/findOne")
	@ResponseBody
	public Group findOne(String  groupId){
		return GroupServiceImpl.findOne(groupId);
	}
	
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */
	@RequestMapping("/delete")
	@ResponseBody
	public Result delete(String [] ids){
		try {
			GroupServiceImpl.delete(ids);
			return new Result(true, "删除成功"); 
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "删除失败");
		}
	}
	
		/**
	 * 查询+分页
	 * @param
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/search")
	@ResponseBody
	public PageResult search(@RequestBody Group group, int page, int rows  ){

		return GroupServiceImpl.findPage(group, page, rows);
	}
	
}
